scripts/httpd.py

```diff
@@ -32,8 +32,17 @@
     print("Usage: httpd.py <directory>", file=sys.stderr)
     sys.exit(1)
 
+server=None
 # Bind to any free port
-server = HTTPServer(('', 0), SimpleHTTPRequestHandler)
+for port in range(40000, 41000):
+    try:
+        server = HTTPServer(('', port), SimpleHTTPRequestHandler)
+    except OSError:
+        continue
+    break
```

Comment:
I think this would be better readable if the ``break`` was moved into an ``else`` branch of the try-except block.